### ISSUE TITLE
xcodebuild -showBuildSettings hangs, Xcode 9 Beta 5

### ISSUE THREAD

**Issue by barrettj** [fastlane/fastlane#9983]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

The xcodebuild -showBuildSettings command seems to hang on the latest Xcode beta.

Command timed out after 10 seconds on try 1 of 4, trying again...
[20:18:59]: Command timed out after 10 seconds on try 2 of 4, trying again...
[20:19:09]: Command timed out after 10 seconds on try 3 of 4, trying again...
[20:19:19]: Command timed out after 10 seconds on try 4 of 4

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

### Captured Output

Command Used: `hockey`
<details><summary>Output/Log</summary>

```

Successfully loaded Appfile at path '/Users/barrettjacobsen/Programming/Attainment/GoVisual/fastlane/Appfile'
- apple_id: 'barrett@attainmentcompany.com'
-------

+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | rm -f                  | 0           |
|      | '/Users/barrettjacobs  |             |
|      | en/Programming/Attain  |             |
|      | ment/GoVisual          |             |
| 2    | increment_build_numbe  | 2           |
|      | r                      |             |
| 3    | commit_version_bump    | 0           |
| 4    | cocoapods              | 4           |
| 💥   | gym                    | 42          |
+------+------------------------+-------------+



```

</details>

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.12.6                                                        |
| Ruby                        | 2.0.0                                                          |
| Bundler?                    | false                                                          |
| Git                         | git version 2.13.3 (Apple Git-91)                              |
| Installation Source         | /usr/local/bin/fastlane                                        |
| Host                        | Mac OS X 10.12.6 (16G29)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib |
| OpenSSL Version             | OpenSSL 0.9.8zc 19 Mar 2015                                    |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/               |
| Xcode Version               | 9.0                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/krausefx/fastlane#customise-the-fastfile

workspace = "GoVisual.xcworkspace"
ipaDestination = Dir.pwd + "/IPAs/"




before_all do
  # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."

  #cleanup IPAs folder
  sh "rm -f '" + ipaDestination + "'*.*"
  
  increment_build_number
  commit_version_bump
  cocoapods 
  #acknowledgePodsNotUPdated = prompt(text: "Pods are not currently being updated when doing a fastlane due to necessary manual modifications!", boolean: true)
end



lane :hockey do
  ENV['APP'] = 'com.attainmentcompany.GoTalk-Scene-Maker'
 
  gym({
    configuration: "Debug",
    scheme: "GoVisual",
    output_directory: ipaDestination,
    export_method: "development",
  })

  releaseNotesPath = File.expand_path(Dir.pwd +  '/../ReleaseNotes.txt')

  doneWithChanges = prompt(text: "Have release notes been updated yet?", boolean: true)

  unless doneWithChanges 
    system('open -Wne "' + releaseNotesPath + '"')
  end

  file = File.open(releaseNotesPath, "rb")
  notes = file.read

  hockey({
    api_token: hockeyApiToken,
    #ipa: Actions.lane_context[Actions::SharedValues::IPA_OUTPUT_PATH],
    notes: notes
  })
end

lane :hockeynotesdone do
  ENV['APP'] = 'com.attainmentcompany.GoTalk-Scene-Maker'
 
  gym({
    configuration: "Debug",
    scheme: "GoVisual",
    output_directory: ipaDestination,
    export_method: "development",
  })

  releaseNotesPath = File.expand_path(Dir.pwd +  '/../ReleaseNotes.txt')

  file = File.open(releaseNotesPath, "rb")
  notes = file.read

  hockey({
    api_token: hockeyApiToken,
    #ipa: Actions.lane_context[Actions::SharedValues::IPA_OUTPUT_PATH],
    notes: notes
  })
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier ENV['APP']

apple_id "barrett@attainmentcompany.com"```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.51.0  | ✅ Up-To-Date  |


### Loaded fastlane plugins:

**No plugins Loaded**


<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.2        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| CFPropertyList            | 2.3.4        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.5        |
| excon                     | 0.54.0       |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| declarative               | 0.0.9        |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| public_suffix             | 2.0.4        |
| addressable               | 2.5.1        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.12.0       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.6       |
| faraday_middleware        | 0.10.1       |
| json                      | 1.8.6        |
| rubyzip                   | 1.2.0        |
| faraday                   | 0.12.1       |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| libxml-ruby               | 2.6.0        |
</details>


*generated on:* **2017-08-07**
</details>

**Comment by robertjpayne** [comment]
Having the same issue -- running it on Command Line doesn't seem to help so I don't think this is a Fastlane issue but an Xcode issue.

**Comment by tizaks** [comment]
I've run into this issue too - I've managed to narrow it down to having a provisioning profile selected for the scheme that this command runs on. I've found if I attempt to run the command after deselecting provisioning profiles (or disabling Automatically manage signing) it works with no issues.

**Comment by waltermvp** [comment]
@tizaks your right that deselecting provisioning profiles helps, I am able to get much farther along in `gym`, however I them get this 👍 
```
❌  Code Signing Error: "myVeeta" requires a provisioning profile. Select a provisioning profile for the "Release" build configuration in the project editor.

[02:37:33]: Exit status: 65
```

**Comment by Ruenzuo** [comment]
I'm also experiencing this. I disabled the timeout setting `FASTLANE_XCODE_LIST_TIMEOUT` to 0 (default is 10 seconds) and I haven't seen the error since.

**Comment by jpsim** [comment]
We filed a radar about this (sorry not on open radar): rdar://33763197

**Comment by mjholgate** [comment]
Ah, so did I oops :) rdar://33778243

**Comment by KrauseFx** [comment]
Thanks everyone, good to hear you all already filed radars, let's hope  hears us and resolves this problem on the next beta. It seems like it's not something we should provide a workaround for in fastlane for now :)

**Comment by lyndsey-ferguson** [comment]
I've added `rdar://33780346`

**Comment by jpsim** [comment]
Probably because it seems to only occur if the development team is set in the project file.

**Comment by lyndsey-ferguson** [comment]
@jpsim what do you mean? Inspired by your comment, I changed `DEVELOPMENT_TEAM` to `""` instead of the Id. Is there something else I can try?

**Comment by jpsim** [comment]
Here's what fixed it in our case: https://github.com/realm/realm-cocoa/pull/5208

**Comment by lyndsey-ferguson** [comment]
Thank you. I removed `DevelopmentTeam` entries and set the `DEVELOPMENT_TEAM` to `""` and that didn't change anything for us.

**Comment by jpsim** [comment]
Make sure to include that in the repro steps for your radar to make sure both our issues are addressed if they happen to be different underlying problems!

**Comment by lyndsey-ferguson** [comment]
I also see this when I leave it running:
```
2017-08-08 17:27:19.254 xcodebuild[25531:2013371] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:19.254 xcodebuild[25546:2013372] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:19.254 xcodebuild[25536:2013370] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:19.254 xcodebuild[25539:2013369] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:19.254 xcodebuild[25514:2013373] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:19.254 xcodebuild[25527:2013374] CoreSimulatorService job is no longer valid. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.510 xcodebuild[25527:2012473] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.510 xcodebuild[25527:2014016] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
2017-08-08 17:27:39.543 xcodebuild[25536:2013415] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.543 xcodebuild[25539:2013477] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.543 xcodebuild[25514:2013466] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.543 xcodebuild[25546:2013412] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.543 xcodebuild[25531:2013436] Failed to load CoreSimulatorService. Access to simulator services will not be available. Error=Error Domain=NSPOSIXErrorDomain Code=61 "Connection refused" UserInfo={NSLocalizedDescription=Unable to lookup com.apple.CoreSimulator.CoreSimulatorService in the bootstrap. This can happen if running with a sandbox profile. When running with a sandbox profile, /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc must be owned by root, not group writable, and not world writable. See <rdar://problem/22142915>.}
2017-08-08 17:27:39.543 xcodebuild[25514:2014028] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
2017-08-08 17:27:39.544 xcodebuild[25536:2014024] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
2017-08-08 17:27:39.544 xcodebuild[25539:2013369] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
2017-08-08 17:27:39.544 xcodebuild[25546:2014029] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
2017-08-08 17:27:39.544 xcodebuild[25531:2014032] CoreSimulatorService connection became invalid. Simulator services will no longer be available.
```

**Comment by lyndsey-ferguson** [comment]
I'll try setting the permissions

**Comment by lyndsey-ferguson** [comment]
It seems that it is owned by root, not group writable, and not world writable:  
```
ls -la /Library/Developer/PrivateFrameworks/CoreSimulator.framework/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc
total 0
drwxr-xr-x 3 root 102 Aug 2 02:49 .
drwxr-xr-x 3 root 102 Aug 2 02:49 ..
drwxr-xr-x 7 root 238 Aug 2 02:50 Contents
```

**Comment by jpsim** [comment]
Not 100% sure, but I don't think those errors are new in Xcode 9, and from what I can tell they're unrelated to this issue.

**Comment by lyndsey-ferguson** [comment]
Thanks @jpsim

**Comment by AsherVo** [comment]
This is an incredibly hacky workaround, but I managed to force the process continue by

1) setting the environment variable FASTLANE_XCODE_LIST_TIMEOUT to 0
2) manually force-quitting the "xcodebuild" process in Activity Monitor

At which point the lane continued.

**Comment by jackymelb** [comment]
**enable automatically manage signing** works for now. Hopefully, it will be fixed in next beta version.

**Comment by apauriche** [comment]
If you set in Xcode > preferences > Locations the command line tools to Xcode 8 it works well but if you use the Xcode 9 beta 5 you'll be stuck.

**Comment by gspiers** [comment]
I've filed a radar as well, 33914240. Hopefully it will be fixed in the next beta.

**Comment by arokiagetz** [comment]
@AsherVo, Can you help how to manually force-quit "xcodebuild" process when it got struck.

@KrauseFx, Please let me know if there is any work-around or how disable to "xcodebuild -showBuildSettings"

Eagerly waiting for a work-around until Apple provides the fix in the next release, as all my builds are getting hung and cant build any app.

**Comment by taquitos** [comment]
@arokiagetz  
If you are running _fastlane_ in an interactive shell, you can also just press `control-c` a few times to kill the process. If that doesn't work, you can open a new terminal and then use: 
`ps aux|grep "xcodebuild"`
the output will show something like: 
`liebowitz         4750   0.0  0.0  2432804   1980 s007  S+   10:11AM   0:00.00 (xcodebuild path)`
then you can do:
`kill -9 4750` (4750 is the process id)
and that should kill the process you find

currently there is no workaround for this bug, other than using Xcode 8, 🤞 that Apple fixes it in the next beta

**Comment by sidmani** [comment]
I tried setting Xcode to Xcode 8.3 with xcode-select and it still hung.

**Comment by mooshee** [comment]
You can kill it with one line like this `kill $(ps aux | grep 'xcodebuild' | awk '{print $2}')`.  I had several of them stuck.

**Comment by AliSoftware** [comment]
Just ran `xcodebuild -showBuildSettings` using today's Xcode9 beta 6 on a simple Xcode project and it seems not to hang anymore.
Haven't tested running `fastlane` with this beta yet though, as I don't have any project with a `Fastfile` on this computer, so I'll have to test on my work computer for that tomorrow. But it's promising (even if no mention of any of this in the Release Notes… 🤔 )!

**Comment by lyndsey-ferguson** [comment]
I have to update my OS version before I can try it, so thank you for giving me hope

**Comment by lyndsey-ferguson** [comment]
@AliSoftware did b5 hang on that same simple project?

**Comment by taquitos** [comment]
Can confirm, it works in B6 for a sample project I used for my radar that didn't work on B5 🎉

**Comment by AliSoftware** [comment]
Sadly can't say, that project is built using SPM and a Rakefile, not Fastlane, as it builds a command line tool (that's my SwiftGen tool) so never tried `xcodebuild -showBuildSettings` in it with b5 before. But all the ipa projects we have at work, building and signing the ipa with Fastlane and all, had the issue with b5. so I'm hoping it doesn't depend that much on the project?
More on that tomorrow once our CI is updated to b6 😉

**Comment by hiedar** [comment]
This problem may be fixed with Xcode9 beta 6.

**Comment by arauchberger** [comment]
issue is solved with Xcode Version 9.0 beta 6 (9M214v)

**Comment by AliSoftware** [comment]
Thanks for the feedback folks
